### DISCUSSION TITLE
Initial shot to test connector

### DISCUSSION THREAD

**Discussion by wonder99** [springbok/MLM2PRO-GSPro-Connector#13] [General]
I'm not sure it's a great idea to provide real shot data in the initial message to the GSP connector--if someone needs to restart your connector for whatever reason, it wouldn't be good to have a random shot entered into their game.  Instead, you can use the heartbeat message for this (with no shot or club data).  I've tried this and it does work

**Comment by springbok** [comment]
Yep agreed, I did use a heartbeat message ordinally but for some reason the gspro connector didn't respond to it even though it was received. So decided to use a real shot to get it going, I will; have another look at it.

**Comment by wonder99** [comment]
True, I don't think there's a response to the heartbeat command but you can
set a socket timeout and if it  expires after sending, you know the
connector is not listening. At least that's what I remember.
Paul

**Reply by springbok** [reply]
Should now be fixed, removed the test shot when it connects, I now assume connection ok if socket connect doesn't return an error.
